packages/react/src/lib/accessibility/Behaviors/Tree/treeItemBehavior.ts

```diff
@@ -30,21 +32,46 @@ const treeItemBehavior: Accessibility<TreeItemBehaviorProps> = props => ({
       performClick: {
         keyCombinations: [{ keyCode: keyboardKey.Enter }, { keyCode: keyboardKey.Spacebar }],
       },
-      collapseOrReceiveFocus: {
-        keyCombinations: [{ keyCode: keyboardKey.ArrowLeft }],
-      },
-      expandOrPassFocus: {
-        keyCombinations: [{ keyCode: keyboardKey.ArrowRight }],
-      },
+      ...(isSubtreeOpen(props) && {
+        receiveFocus: {
+          keyCombinations: [{ keyCode: keyboardKey.ArrowLeft }],
+          shouldHandle: doesEventComesFromChildItem,
+        },
+        collapse: {
+          keyCombinations: [{ keyCode: keyboardKey.ArrowLeft }],
+          shouldHandle: event => !doesEventComesFromChildItem(event),
+        },
+      }),
+      ...(!isSubtreeOpen(props) && {
```

Comment:
I am not sure I like the combination of conditionally **adding** key handlers based on `props` and conditionally **invoking** them based on `event` (and eventually `props`).
Wouldn't it be easier to switch everything to `shouldHandle` condition? (I am aware of the fact that this can result in added keyhandlers which never call any action.)
Other opinions?

Reply:
@miroslavstastny I totally share your thought, have same concerns. 
@kuzhelov @mnajdova @layershifter need your input here